File: src/types/review.ts
export interface DiffLine {
  type: "add" | "remove" | "context" | "hunk"
  line: number | null
  text: string
}

export interface ReviewIssue {
  file: string | null
  line: number | null
  description: string
  severity: "High" | "Medium" | "Low" | "NEEDS_VERIFICATION"
  category: "Code Quality" | "Logic" | "Stability" | "Performance" | "Security" | "Best Practices"
  suggestedComment: string
  selected: boolean
  diffHunk?: DiffLine[]
}

export interface PRReview {
  pr: import("./pr").PullRequest
  verdict: "APPROVE" | "REQUEST_CHANGES" | "NEEDS_DISCUSSION"
  summary: string
  issues: ReviewIssue[]
  positiveNotes: string[]
  overallNotes: string
  reviewedAt: string
  engine: string
  diffMap: Record<string, Record<number, number>>
  commitSha: string
}

export interface ReportMeta {
  id: string
  runAt: string
  prCount: number
  engine: string
}

export interface Report {
  id: string
  runAt: string
  reviews: PRReview[]
  engine: string
}

File: src-tauri/src/models/review.rs
use serde::{Deserialize, Serialize};
use crate::models::pr::PullRequest;
use std::collections::HashMap;

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DiffLine {
    pub r#type: String,  // "add" | "remove" | "context" | "hunk"
    pub line: Option<u32>,
    pub text: String,
}

#[derive(Debug, Clone, Serialize, Deserialize, Default)]
#[serde(rename_all = "camelCase")]
pub struct ReviewIssue {
    pub file: Option<String>,
    pub line: Option<u32>,
    pub description: String,
    pub severity: String,
    pub category: String,
    pub suggested_comment: String,
    pub selected: bool,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub diff_hunk: Option<Vec<DiffLine>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PRReview {
    pub pr: PullRequest,
    pub verdict: String,
    pub summary: String,
    pub issues: Vec<ReviewIssue>,
    pub positive_notes: Vec<String>,
    pub overall_notes: String,
    pub reviewed_at: String,
    pub engine: String,
    pub diff_map: HashMap<String, HashMap<u32, u32>>,
    pub commit_sha: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ReportMeta {
    pub id: String,
    pub run_at: String,
    pub pr_count: usize,
    pub engine: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Report {
    pub id: String,
    pub run_at: String,
    pub reviews: Vec<PRReview>,
    pub engine: String,
}
